Document composition behavior for executable directives (#2333)
apollographql/federation: docs/source/federated-types/composition.mdx

Therefore, the definition of the `Color` enum **must match exactly** in every subgraph that defines it. An exact match is the only scenario that enables union _and_ intersection to produce the same result.

The subgraph schemas above _do not compose_, because their definitions of the `Color` enum differ.

## Directives

> Composition will handle a directive differently depending on whether it is an "executable" directive or a "type system" directive.

Reviewer: ```suggestion
Composition handles a directive differently depending on whether it's an "executable" directive or a "type system" directive.
```

> ### Executable Directives

Reviewer: ```suggestion
### Executable directives
```

> Executable directives are intended to be used by the client in a query. They are defined on one or more of the [executable directive locations](http://spec.graphql.org/June2018/#ExecutableDirectiveLocation). For example, you might have a directive definition of `directive @lowercase on FIELD` which a client could use in their query like this:

Reviewer: ```suggestion
Executable directives are intended to be used by clients in their queries. They are applied to one or more of the [executable directive locations](http://spec.graphql.org/June2018/#ExecutableDirectiveLocation). For example, you might have a directive definition of `directive @lowercase on FIELD`, which a client could use in their query like so:
```

```graphql
query {
  getSomeData {
    someField @lowercase
  }
}
```

> These types of directives are composed into the supergraph schema only if _all_ of these conditions are met:

Reviewer: ```suggestion
An executable directive is composed into the supergraph schema only if _all_ of the following conditions are met:
```

> - The directive is defined in _all_ subgraphs
> - The directive is defined in the same way in _all_ subgraphs
> - The directive _is not_ used in any [`@composeDirective`](./federated-directives#composedirective) directives

Reviewer: ```suggestion
- The directive is defined in _all_ subgraphs.
- The directive is defined _identically_ in _ all subgraphs.
- The directive is _not_ included in any [`@composeDirective`](./federated-directives#composedirective) directives.
```

Author: Should `all` still be italicized or no?

Reviewer: ah whoops, nah, just "identically" here, now that "all subgraphs" has been emphatically established in the previous bullet

> ### Type System Directives

Reviewer: ```suggestion
### Type system directives
```

> Type system directives help define the structure of the schema and are not intended for use by clients. They include any directive which is defined on one or more of the [type system directive locations](http://spec.graphql.org/June2018/#TypeSystemDirectiveLocation) These directives will not be composed into the supergraph schema, but can be used to provide information to the router via [`@composeDirective`](./federated-directives#composedirective) directive.

Reviewer: ```suggestion
Type system directives help define the structure of the schema and are not intended for use by clients. They are applied to one or more of the [type system directive locations](http://spec.graphql.org/June2018/#TypeSystemDirectiveLocation).

These directives are _not_ composed into the supergraph schema, but they can still provide information to the router via the [`@composeDirective`](./federated-directives#composedirective) directive.
```
